### PR TITLE
core: better import collision error

### LINKED ISSUE #849: Have a better error for import collisions

### PR DESCRIPTION
**Pull request by SmailBarkouch**
fixes #849

### CODE REVIEW

File: clients/linux/src/backend.rs [hunk 1]
```diff
@@ -30,20 +30,20 @@ macro_rules! match_core_err {
     (
         $err:expr,
         $enum:ident,
-        $( $variants:ident => $matches:expr ),+,
+        $( $variants:ident $( ( $val:ident ) )? => $matches:expr ),*,
```
Comment (steverusso) [inline clients/linux/src/backend.rs:33]: Why the change from `+` to `*`?

Comment (SmailBarkouch) [reply clients/linux/src/backend.rs]: I was experimenting and forgot to change it back. My bad.

Comment (steverusso) [reply clients/linux/src/backend.rs]: Ah ok :+1:

File: core/src/service/import_export_service.rs [hunk 2]
```diff
@@ -76,7 +80,13 @@ fn import_file_recursively(
             fs::read_dir(disk_path).map_err(CoreError::from)?.collect();
 
         if children.is_empty() {
-            path_service::create_at_path(config, &lockbook_path_with_new)?;
+            path_service::create_at_path(config, &lockbook_path_with_new).map_err(|err| {
+                if err == CoreError::PathTaken {
```
Comment (steverusso) [inline core/src/service/import_export_service.rs:84]: nit: I think a match would be cleaner here.

File: core/src/service/import_export_service.rs [hunk 3]
```diff
@@ -80,13 +80,13 @@ fn import_file_recursively(
             fs::read_dir(disk_path).map_err(CoreError::from)?.collect();
 
         if children.is_empty() {
-            path_service::create_at_path(config, &lockbook_path_with_new).map_err(|err| {
-                if err == CoreError::PathTaken {
-                    CoreError::ImportCollision(lockbook_path_with_new)
-                } else {
-                    err
+            match path_service::create_at_path(config, &lockbook_path_with_new) {
```
Comment (steverusso) [inline core/src/service/import_export_service.rs:83]: I don't understand the change here, why not just match within the map_err?

Comment (steverusso) [reply core/src/service/import_export_service.rs:83]: The `map_err` that was there previously: https://github.com/lockbook/lockbook/pull/856#discussion_r693281304

Comment (SmailBarkouch) [reply core/src/service/import_export_service.rs:83]: Oh, I thought you meant match the entire thing instead of map_err. I was confused too, but now it makes sense haha.

**Review by Parth — APPROVED** [review]
